server/src/useCases/report/productCompatibility/utils/filtersValidation.js

```diff
@@ -0,0 +1,7 @@
+module.exports = class FiltersValidation {
+  static validateFilter(filter, defaultValue) {
+    return filter === undefined || !filter.match(/^\d+$/)
+      ? defaultValue
+      : filter;
+  }
```

Comment:
I don't think we should allow Infinity on regular calls. 
Can you add check for max limit (either configurable or fixed on 1k) just for safety measures?

Reply:
Fixed, please confirm if the solution meets your expectations @dborowiecki